### ISSUE TITLE
Apple Silicon native support

### ISSUE THREAD

**Issue by Shanduur** [tailscale/tailscale#943]
**Is your feature request related to a problem? Please describe.**

Apple Silicon build, when Go releases native support. 

**Describe the solution you'd like**

Run TailScale on Apple Silicon natively.

**Describe alternatives you've considered**

Running it through Rosetta.

**Additional context**

Apple just released macs with their custom SOC. They are based on ARM architecture. Go has no support for it yet, but when build supporting it arrives, it would be wise to compile TailScale for it ASAP.

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_2rrkh)

**Comment by kivikakk** [comment]
Fwiw, I've tried building Tailscale with the [Go 1.16 beta on Apple Silicon](https://tip.golang.org/doc/go1.16), and at the very least the `tailscale` binary works well. I don't quite know how to get `tailscaled` running on MacOS without the help of the GUI.

I had to hack a bunch of `+build` directives since we now have `darwin,arm64`, but here we are (running the App Store daemon, non-private IPs manually redacted):

```
tapioca ~/C/tailscale ((v1.2.10)|✚4) $ go install tailscale.com/cmd/tailscale{,d}
tapioca ~/C/tailscale ((v1.2.10)|✚4) $ file ~/Code/gopath/bin/tailscale
/Users/kameliya/Code/gopath/bin/tailscale: Mach-O 64-bit executable arm64
tapioca ~/C/tailscale ((v1.2.10)|✚4) [1] $ ~/Code/gopath/bin/tailscale status
[twQNK]         w.x.y.z  tapioca.tomodachi  tx=       0 rx=       0  syd  a.b.c.d:41641, 192.168.1.120:41641
[CgEoz] linux   w.x.y.z  raven              tx=       0 rx=       0  syd  a.b.c.d:1028, 192.168.1.60:41641, 192.168.122.1:41641, 172.17.0.1:41641
[GNIIn] linux   w.x.y.z  muffin             tx=       0 rx=       0  syd  a.b.c.d:1026, 192.168.1.168:41641, 192.168.1.175:41641
[ik4vt] freebsd e.f.g.h  talia              tx=   15140 rx=   84252  syd  i.j.k.l:41641, 10.0.1.4:41641
[…]
```

**Comment by eugene1g** [comment]
I'm running Tailscale on Apple Silicon with Rosetta 2 - no issues. Didn't have to do anything special, just installed & ran it exactly as I did on my main Intel Mac. Rosetta 2 is awesome (and so is Tailscale ;) )

**Comment by DentonGentry** [comment]
Quite a few users are running the current Tailscale packages via Rosetta x86 emulation (including me). Go 1.16 is supposed to be released some time this month, when we move to that version of the toolchain a native binary seems feasible.

**Comment by bradfitz** [comment]
Tracking bug for switching to Go 1.16: #1370

**Comment by bradfitz** [comment]
As an update here: this is all working on M1 Macs and will be released in about 2 weeks when Tailscale 1.6 comes out.

**Comment by arthurlockman** [comment]
I just upgraded to 1.6 from the MAS. Seems like it's still Intel?
<img width="377" alt="image" src="https://user-images.githubusercontent.com/158981/111850964-13bee580-88e8-11eb-9d9d-039bc2043959.png">

**Comment by bradfitz** [comment]
Yeah, uh, we clicked the wrong button in xcode. 😬 We'll fix it for 1.6.1.

**Comment by apenwarr** [comment]
This was fixed in v1.8.
